presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -124,6 +125,21 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
         addJoinsWithDifferentDistributions(joinNode, possibleJoinNodes, context);
         addJoinsWithDifferentDistributions(joinNode.flipChildren(), possibleJoinNodes, context);
 
+        if (isBelowMaxBroadcastSize(joinNode, context) && isBelowMaxBroadcastSize(joinNode.flipChildren(), context)) {
```

Comment:
consider how to incorporate this with the size based join paths.  I also feel that this rule now has too many branches for how we choose which table to broadcast (not just from your change, there just keep being more heurisitics added) and you end up in these mutually exclusive branches where either you need to repeat code to e.g. choose the highest confidence stats from multiple branches of the code, or you only get that heuristic if you go down one branch but not the other. It would be nice to consolidate them.